### PR TITLE
Add toml configuration file support

### PR DESCRIPTION
**Pull request by Lumexralph**
#### Description

Setup was initially done using the command line flags, to make it easy to setup configuration from a file, this PR adds that support.

### CODE REVIEW

File: config.toml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[session]
+rooms = ["room 1", "room 2"]
```
Comment (tarrencev) [inline config.toml]: lets make this an integer input. so i can just say `2`  and it will generate a room name like `room %d`

File: main.go [hunk 2]
```diff
@@ -73,25 +118,42 @@ func main() {
 	var stagger float64
 
 	flag.StringVar(&input, "input", "", "path to the input media")
-	flag.StringVar(&sfu, "sfu", "localhost:50051", "ion-sfu grpc url")
+	flag.StringVar(&sfu, "sfu", "50051", "ion-sfu grpc url")
 	flag.Var(&rooms, "room", "Rooms to join.")
```
Comment (tarrencev) [inline main.go]: Lets remove the existing cli args and just support config.toml for now. later we can support overloading with cli flags

File: main.go [hunk 3]
```diff
@@ -73,25 +118,42 @@ func main() {
...
 	flag.BoolVar(&audio, "audio", false, "Publish audio stream from webm file")
 	flag.BoolVar(&produce, "produce", false, "path to the media file you want to playback")
 	flag.BoolVar(&consume, "consume", false, "Run subscribe to all streams and consume data")
+	flag.StringVar(&file, "c", "", "config file")
 
 	flag.Parse()
 
+	if loaded := loadConfig(); loaded {
+		log.Println("configuration successfully loaded!")
+		rooms = conf.Rooms
```
Comment (tarrencev) [inline main.go]: remove all these vars and just read directly off `conf`

File: ion/client.go [hunk 4]
```diff
@@ -38,17 +38,19 @@ type LoadClient struct {
...
-	log.Printf("Creating load client => name: %s room: %s input: %s", name, room, input)
-
+	log.Printf("Creating load client => name: %s room: %s input: %s address: %s\n", name, room, input, address)
 	// Set up a connection to the sfu server.
 	conn, err := grpc.Dial(address, grpc.WithInsecure(), grpc.WithBlock())
 	if err != nil {
 		log.Fatalf("did not connect: %v", err)
 	}
+	log.Println("connected to grpc: ", input)
 	c := sfu.NewSFUClient(conn)
```
Comment (tarrencev) [inline ion/client.go:48]: in a followup PR, we should reuse the SFUClient. We dont need a new instance for each client

**Review by tarrencev — CHANGES_REQUESTED** [review]

File: config.toml (hunk 1 above)
Comment (Lumexralph) [reply config.toml]: makes sense, thought of that but didn't want to break what already existed.

File: ion/client.go (hunk 4 above)
Comment (Lumexralph) [reply ion/client.go:48]: I agree.

**Review by tarrencev — APPROVED** [review]

**Pull request closed without merging**